aptos-move/block-executor/src/scheduler.rs

```diff
@@ -16,7 +16,10 @@ use std::{
 pub type TxnIndex = usize;
 pub type Incarnation = usize;
 pub type Version = (TxnIndex, Incarnation);
-type DependencyCondvar = Arc<(Mutex<bool>, Condvar)>;
+// The first boolean indicates whether the dependency is resolved.
```

Comment:
Let's make an enum.
DependencyStatus:
We can have: 
Unresolved, // dependency is not resolved yet.
Resolved, // dependency is resolved
Terminated, // the block execution is completed